src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -219,117 +207,103 @@ def _entity_ids_by_type(self,
 
 @attr.s(frozen=True, kw_only=True, auto_attribs=True)
 class GitHubStagingAreaFactory:
-    repo: Repository
-    ref: str
-    path: Tuple[str, ...]
+    url: str
+    repo: git.Repo
+    commit: git.Commit
+    path: tuple[str, ...]
+    tree: git.Tree
 
     @classmethod
     def create(cls,
-               owner: str,
-               name: str,
+               url: str,
                ref: str,
-               path: Optional[str] = None
+               path: tuple[str, ...],
+               tmpdir: str
                ) -> 'GitHubStagingAreaFactory':
         """
-        :param owner: The owner of the GitHub repository
-        :param name: The name of the GitHub repository
+        :param url: The remote URL of a Git repository on GitHub
         :param ref: A branch name, tag, or commit SHA
-        :param path: The path inside the repository to the base of the staging area
+        :param path: The path inside the repository to the base of the staging
+                     area
+        :param tmpdir: The path to a local directory where the repository will
+                       be cloned
+
         """
-        token = os.environ['GITHUB_TOKEN']  # A GitHub personal access token
-        log.debug('Requesting GitHub repo %s', (owner, name, ref, path))
-        github_api = Github(token)
-        rate_limit = github_api.get_rate_limit()
-        log.debug('GitHub API rate limit limit=%i, remaining=%i, reset=%s',
-                  rate_limit.core.limit, rate_limit.core.remaining, rate_limit.core.reset)
-        repo = github_api.get_repo(f'{owner}/{name}')
-        path = () if path == '' else tuple(path.split('/'))
-        return cls(repo=repo, ref=ref, path=path)
+        log.debug("Cloning '%s' into '%s'", url, tmpdir)
+        repo = git.Repo.clone_from(url, tmpdir)
+        commit = repo.commit(ref)
+        tree = repo.tree(commit)
+        tree = cls._get_subtree(tree, path)
```

Comment:
Temporary assignment doesn't seem helpful to me
```suggestion
        tree = cls._get_subtree(repo.tree(commit), path)
```